Automattic/wp-calypso
# Playwright: Part 2. Migrate rest of wp-likes spec and write additional cases.

#### Changes proposed in this Pull Request

This PR migrates rest of the [`wp-likes-spec.js`](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/specs-calypso/wp-likes-spec.js) file from the Selenium suite.

As part of the migration, the test cases have been split up where possible:
- `wp-likes__comments-spec.js` test the likes feature for comments only.
- `wp-like__posts-spec.js` test the likes feature for posts only.

The rubric of cases covered by this migration looks like the following:

Like and Unlike
| | posts | comments |
| --- | --- | --- |
| new | ✓ | ✓ |
| existing |✓ | ✓ |

To implement the Likes tests, additional supporting library files have also been migrated/created where applicable.

- `CommentComponent` represents the comment section of a post. 
- `PublishedPostPage` represents the post.



#### Testing instructions

CI
- ensure tests pass without any issues on TeamCity.

Local
- pull the branch and ensure they run with `mocha --config .mocharc_playwright.yml specs/specs-playwright/`.

Likes: Comment should look like this:
```
  [WPCOM] Likes (Comment): (desktop) @parallel
    Comment and like on an existing post
Logging in as e2eflowtestinggutenbergsimple
      ✓ Log in (3065ms)
      ✓ Visit site (1191ms)
      ✓ Click on first post (254ms)
      ✓ Post a comment (3712ms)
      ✓ Like a comment (1897ms)
    Comment and like on a new post
Logging in as e2eflowtestinggutenbergsimple
      ✓ Log in (2790ms)
      ✓ Start new post (5595ms)
      ✓ Enter post title (261ms)
      ✓ Enter post text (670ms)
      ✓ Publish and visit post (6067ms)
      ✓ Post a comment (3262ms)
      ✓ Like a comment (1994ms)
      ✓ Unlike a comment (616ms)


  13 passing (38s)
```

Likes: Post should look like this:
```
  [WPCOM] Likes (Post): (desktop) @parallel
    New post
Logging in as e2eflowtestinggutenbergsimple
      ✓ Log in (3007ms)
      ✓ Start new post (6651ms)
      ✓ Enter post title (330ms)
      ✓ Enter post text (676ms)
      ✓ Publish and visit post (5540ms)
      ✓ Like post (1080ms)
      ✓ Unlike post (96ms)
    Existing post
Logging in as e2eflowtestinggutenbergsimple
      ✓ Log in (2724ms)
      ✓ Visit site (849ms)
      ✓ Click on first post (321ms)
      ✓ Like post (1246ms)


  11 passing (34s)
```

Parent: #53169
